libs/lilxml.c

```diff
@@ -146,8 +147,9 @@ void lilxmlMalloc(void *(*newmalloc)(size_t size), void *(*newrealloc)(void *ptr
 /* pass back a fresh handle for use with our other functions */
 LilXML *newLilXML()
 {
-    LilXML *lp = (LilXML *)moremem(NULL, sizeof(LilXML));
-    memset(lp, 0, sizeof(LilXML));
+    LilXML *lp = (LilXML *)moremem(NULL, sizeof *lp);
+    memset(lp, 0, sizeof *lp);
+    assert(lp);
```

Comment:
Your assert is executed too late. Instead, could you add a bailout path instead when moremem return null?